server/http/src/service.rs

```diff
@@ -11,6 +13,40 @@ use hyper::{Body, Method, Request, Response, StatusCode};
 use crate::request::GraphQLRequest;
 use crate::response::GraphQLResponse;
 
+pub struct GraphQLServiceMetrics {
+    query_execution_time: Box<HistogramVec>,
+}
+
+impl fmt::Debug for GraphQLServiceMetrics {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "GraphQLServiceMetrics {{ }}")
+    }
+}
+
+impl GraphQLServiceMetrics {
+    pub fn new<M: MetricsRegistry>(registry: Arc<M>) -> Self {
+        let query_execution_time = registry
+            .new_histogram_vec(
+                format!("subgraph_query_execution_time"),
```

Comment:
I wonder if we should shorten this to `query_time`, like in the logs.

Reply:
Want to go with the shorter, `query_time`?  
I vacillated between the two options myself and ended up using `query_execution_time`  because I think it's more clear. In the logs we use `query_time_ms`;  I think aside from being useful to express the unit of measurement,  the `_ms` suffix makes the distinction between execution time and time of execution clear.

Reply:
Ok, let's keep the longer version. We'll want to add more detailed metrics for query execution at some point. We'll see about the names again then.